### PR TITLE
MINOR; Move quota integration tests to using the new quota API.

### PR DESCRIPTION
**Pull request by dajac**
This PR refactors the various quota integration tests in the `kafka.api` package and migrates them to using the new quota API instead of writing the quota to ZK directly.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dajac** [comment]
@ijuma @rajinisivaram Could you review this one when you have time?

File: core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala [hunk 1]
```diff
@@ -335,12 +339,31 @@ abstract class QuotaTestClients(topic: String,
       assertEquals("Should not have been throttled", 0.0, metricValue(maxMetric), 0.0)
   }
 
-  def quotaProperties(producerQuota: Long, consumerQuota: Long, requestQuota: Double): Properties = {
-    val props = new Properties()
-    props.put(DynamicConfig.Client.ProducerByteRateOverrideProp, producerQuota.toString)
-    props.put(DynamicConfig.Client.ConsumerByteRateOverrideProp, consumerQuota.toString)
-    props.put(DynamicConfig.Client.RequestPercentageOverrideProp, requestQuota.toString)
-    props
+  // Default is set by using Some(null) for either or both user and clientId
```
Comment (rajinisivaram) [inline core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala]: Maybe add a constant for the default entity and use `Some(DefaultEntity)` which would avoid the need for the comment that sounds like we are doing something we shouldn't be.

**Review by rajinisivaram — APPROVED** [review]
@dajac Thanks for the PR. Left one minor comment, apart from that LGTM.

**Comment by dajac** [comment]
@rajinisivaram Thanks for the review. I just pushed an updated that addressed your comment.

**Comment by rajinisivaram** [comment]
ok to test

**Comment by rajinisivaram** [comment]
Connect test failure unrelated, merging to trunk.

**Pull request merged**